src/riak_kv_qry.erl

```diff
@@ -118,8 +257,10 @@ maybe_await_query_results(_) ->
     % we can't use a gen_server call here because the reply needs to be
     % from an fsm but one is not assigned if the query is queued.
     receive
-        Result ->
-            Result
+        {ok, Result} ->
```

Comment:
It's dialyzer wisperings. Because this function returns whatever it receives, dialyzer has no way of knowing what the return type would be. By constraining the terms on reception to only be one of `{ok, Result}` or `{error, Reason}`, we give dialyzer a useful hint.